core/shared/src/main/scala/zio/Hub.scala

```diff
@@ -378,32 +399,30 @@ object Hub {
 
       def handleSurplus(
         hub: internal.Hub[A],
-        subscribers: Set[(internal.Hub.Subscription[A], MutableConcurrentQueue[Promise[Nothing, A]])],
+        subscribers: java.util.Set[(internal.Hub.Subscription[A], MutableConcurrentQueue[Promise[Nothing, A]])],
         as: Iterable[A],
         isShutDown: AtomicBoolean
       )(implicit trace: Trace): UIO[Boolean] =
         ZIO.fiberIdWith { fiberId =>
-          val promise = Promise.unsafe.make[Nothing, Boolean](fiberId)(Unsafe.unsafe)
+          val promise = Promise.unsafe.make[Nothing, Boolean](fiberId)(Unsafe)
           ZIO.suspendSucceed {
             unsafeOffer(as, promise)
             unsafeOnHubEmptySpace(hub, subscribers)
             unsafeCompleteSubscribers(hub, subscribers)
             if (isShutDown.get) ZIO.interrupt else promise.await
-          }.onInterrupt(ZIO.succeed(unsafeRemove(promise)))
+          }.onInterrupt(Exit.succeed(unsafeRemove(promise)))
```

Comment:
I think you should keep `ZIO.succeed` here. I don't think we should rely on the by-name suspension in this case.

```suggestion
          }.onInterrupt(ZIO.succeed(unsafeRemove(promise)))
```